hooks/test/browser/useMemo.test.js

```diff
@@ -38,4 +38,51 @@ describe('useMemo', () => {
 		expect(results).to.deep.equal([2, 2, 3, 3]);
 		expect(memoFunction).to.have.been.calledTwice;
 	});
+
+	it('short circuits diffing for memoized components', () => {
+		let spy = sinon.spy();
+		let spy2 = sinon.spy();
+		const X = ({ count }) => {
+			spy();
+			return <span>{count}</span>;
+		};
+
+		const Y = ({ count }) => {
+			spy2();
+			return <p>{count}</p>;
+		};
+
+		const App = ({ x }) => {
+			const y = useMemo(() => <Y count={x} />, [x]);
+			return (
+				<div>
+					<X count={x} />
+					{y}
```

Comment:
This will only change reference once the `props.x` changes and is a strictly equal vnode for that matter